### PR TITLE
specification cleanup

### PR DESCRIPTION
**Pull request by davidiw**
this is largely a content tweak to use better names and articulate concepts better.

### CODE REVIEW

File: specifications/transactions/README.md [hunk 1]
```diff
@@ -4,21 +4,19 @@
...
 
 ## Introduction
 
 ### On-Chain
 
-A Diem transaction consists of a sender, a script, and a set of arguments. Currency transfer transactions contain a set of arguments including a recipient, an amount, a currency, and metadata.  Diem leverages the metadata to provide context behind a transaction. However, the metadata itself is not validated by the Diem blockchain or Move standard library.
+A Diem transaction consists of a sender, a script, and a set of arguments. Currency transfer transactions arguments include a recipient, an amount, a currency, and metadata. Diem leverages the metadata to provide context behind a transaction. The metadata itself is not validated by the Diem blockchain or Move standard library.
```
Comment (sunmilee) [inline specifications/transactions/README.md:17]:
```suggestion
A Diem transaction consists of a sender, a script, and a set of arguments. Currency transfer transaction arguments include a recipient, an amount, a currency, and metadata. Diem leverages the metadata to provide context behind a transaction. The metadata itself is not validated by the Diem blockchain or Move standard library.
```

File: specifications/transactions/README.md [hunk 2]
```diff
@@ -33,12 +31,12 @@ Diem defines refunds as part of the transaction framework. The refund includes a
 ### Transfer with Subaddresses
 
 Bob would like to send Alice 10 XUS:
-* Alice loads her VASP's mobile APP and generate a PaymentURI
-* The PaymentURI contains a subaddress that uniquely identifies Alice along with additional parameters specifying sending 10 XUS
+* Alice loads her VASP's mobile APP and generate an intent identifier
```
Comment (sunmilee) [inline specifications/transactions/README.md:34]:
```suggestion
* Alice loads her VASP's mobile APP and generates an intent identifier
```

File: specifications/transactions/README.md [hunk 3]
```diff
@@ -33,12 +31,12 @@ Diem defines refunds as part of the transaction framework. The refund includes a
...
+* The intent identifier contains a subaddress that uniquely identifies Alice along with additional parameters specifying sending 10 XUS
 * Alice shares this with Bob via a QR code
-* The PaymentURI opens Bob's VASP's mobile app and confirms Bob's intent to send 10 XUS
+* The intent identifier opens Bob's VASP's mobile app and confirms Bob's intent to send 10 XUS
```
Comment (sunmilee) [inline specifications/transactions/README.md:37]: Does novi's flow look like this? intent identifier links to the novi app and confirms?

File: specifications/transactions/onchain/README.md [hunk 4]
```diff
@@ -1,10 +1,10 @@
...
+Diem transactions mutate and create state (or resources) within the set of [on-chain modules](https://github.com/diem/diem/tree/main/language/diem-framework/modules), primarily the [Diem Account](https://github.com/diem/diem/blob/main/language/diem-framework/modules/doc/DiemAccount.md). The transaction format is defined in the [Move Adapter Specification](https://github.com/diem/diem/blob/main/specifications/move_adapter/README.md). Most participants of the Diem Payment Network (DPN) will submit SignedTransactions containing a script. Diem stores the scripts in [compiled form](https://github.com/diem/diem/tree/main/language/diem-framework/compiled/legacy/transaction_scripts) and in their [original form](https://github.com/diem/diem/tree/main/language/diem-framework/transaction_scripts). The DPN MainNet only allows this set of pre-registerd scripts to be submitted. Due to the evolving nature of Move and the Move compiler, compiling existing scripts may not result in the form stored in the directory stored above. Hence, it is recommended to use the compiled scripts.
```
Comment (sunmilee) [inline specifications/transactions/onchain/README.md:3]: `https://github.com/diem/diem/tree/main/language/diem-framework/transaction_scripts` url says page not found

**Review by sunmilee — APPROVED** [review]

**Comment by aching** [comment]
@sunmilee has a good eye!  Looks good otherwise.

**Review by xli — APPROVED** [review]

**Pull request merged**